src/workerd/jsg/jsg.h

```diff
@@ -2325,6 +2328,21 @@ class ExternalMemoryAdjustment final {
   void maybeDeferAdjustment(ssize_t amount);
 };
 
+// ======================================================================================
+// An abstraction of a virtual file system node.
+class Url;
+namespace fs {
+enum class Node_Type {
+  FILE,
+  DIRECTORY,
+};
+
+class File;
```

Comment:
Redundant forward declaration?
esp if we move filesystem out of the already overloaded jsg layer.